File: Unihub_app/lib/models/faculty.dart
class Faculty {
  final String name;
  final List<dynamic> degrees;

  Faculty(
    this.name,
    this.degrees,
  );

  factory Faculty.fromMap(Map<String, dynamic> json) {
    return Faculty(
      json['name'],
      json['degrees'],
    );
  }
}

File: backend/src/models/Faculty.ts
import {Schema,model,Document} from 'mongoose';
import Degree from './Degree';

const schema = new Schema({
    name: String,
    degrees: Array,
}, {collection: 'faculties'});

interface IFaculty extends Document {
    name: String;
    degree: Array<typeof Degree>;
}

export default model<IFaculty>('Faculty',schema);